sdk/data/azcosmos/emulator_cosmos_gem_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package azcosmos
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGlobalEndpointManager(t *testing.T) {
+	emulatorTests := newEmulatorTests(t)
```

Comment:
This is laid down as an Emulator tests, but there are cases (like multiple write locations) that really cannot be tested with the Emulator.

The Global Endpoint Manager can be tested through unit tests, without the Emulator, you can mock HTTP responses (for example, mock an HTTP response that contains multiple write and read regions and has the EnableMultipleWriteLocations property true or false) and verify that the GEM is loading it correctly.

For example: 

https://github.com/Azure/azure-sdk-for-go/blob/c39ac95a0bce6890c1f3c4403b8b0e2a5d6be2f0/sdk/data/azcosmos/cosmos_client_test.go#L564-L587

You can create mock TLS Server with an specific HTTP JSON response and use it to create a pipeline and a client to inject into the GEM constructor:

```
srv, close := mock.NewTLSServer()
defer close()
srv.AppendResponse(
	mock.WithBody("...the raw account properties json..."),
	mock.WithStatusCode(200))

pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{}, &policy.ClientOptions{Transport: srv})
client := &Client{endpoint: srv.URL(), pipeline: pl}
```

And use that to verify the methods, and even the GetAccountProperties calls.

Reply:
okay, do we want an emulator test at all?

Reply:
I think it's not a matter of want, but rather, would emulator tests cover all the possible scenarios? If the answer is no, then probably that clarifies if we should be using emulator or unit tests. I was able to at least find some scenarios where the emulator test would not cover it, unless you can find a way for the emulator to yield an account with multiple write locations (I personally don't know how but I'm open if that's possible).